src/literal.h

```diff
@@ -31,6 +31,10 @@ namespace wasm {
 
 class Literals;
 struct ExceptionPackage;
+struct GCData;
+// Subclass the vector type so that this is not easily confused with a vector of
+// types (which could be confusing on the Literal constructor).
+struct RttSupers : std::vector<Type> {};
```

Comment:
I don't prefer the superclass pattern in general. But here very specifically, not using a superclass seems dangerously ambiguous, as the Literal constructor would take a array of Types - that seems pretty generic, and not obviously the contructor of  an Rtt. So I think it's good to superclass here?

Reply:
We would want the Literal constructor to still take an `RttSupers`, but `RttSupers` would _contain_ a `std::vector<Type>` rather than _being_ a `std::vector<Type>`. This seems cleaner to me than subclassing classes from the standard library, but I'm not sure why exactly.

Reply:
Interesting, I don't have a preference myself between subclassing std::vector or containing one. But let's discuss separately as you said.